examples/complex.c

```diff
@@ -80,6 +81,52 @@ static int certfail_handler(const xmpp_tlscert_t *cert,
     return read_char[0] == 'y' || read_char[0] == 'Y';
 }
 
+static int
+password_callback(char *pw, size_t pw_max, const char *fname, void *userdata)
+{
+    (void)userdata;
+    /* when using an encrypted keyfile, we get asked multiple times for the
+     * password ... cache it until we're connected
+     */
+    static struct {
+        char pass[256];
+        unsigned char hash[XMPP_SHA1_DIGEST_SIZE];
+        size_t passlen, fnamelen;
+        unsigned count;
+    } password_cache;
+    int ret = -1;
+    unsigned char hash[XMPP_SHA1_DIGEST_SIZE];
+
+    size_t fname_len = strlen(fname);
+    xmpp_sha1_digest((void *)fname, fname_len, hash);
+    if (fname_len && fname_len == password_cache.fnamelen &&
+        memcmp(hash, password_cache.hash, sizeof(hash)) == 0) {
+        if (password_cache.passlen && --password_cache.count) {
+            memcpy(pw, password_cache.pass, password_cache.passlen + 1);
+            ret = password_cache.passlen;
+            if (!password_cache.count)
+                memset(&password_cache, 0, sizeof(password_cache));
+            return ret;
+        }
+    }
+    printf("Trying to unlock %s\n", fname);
+    char *pass = getpass("Please enter password: ");
+    if (!pass)
+        return -1;
```

Comment:
you could also return `ret` here.
Or use `if (pass)` and nest the next if cluase into this one, and have just one return at the end (since you are doing it this way in the next if clause).